althea_kernel_interface/src/exit_server_tunnel.rs

```diff
@@ -76,8 +90,34 @@ impl dyn KernelInterface {
         Ok(())
     }
 
+    /// This function adds a route for each client subnet to the ipv6 routing table
+    /// through wg_exit
+    pub fn setup_client_routes(&self, client_ipv6_list: String) {
+        if client_ipv6_list.is_empty() {
+            return;
+        }
+
+        let ipv6_list: Vec<&str> = client_ipv6_list.split(',').collect();
+
+        for ip in ipv6_list {
+            // Verfiy its a valid subnet
+            if let Ok(ip_net) = ip.parse::<IpNetwork>() {
+                let _res = self.run_command(
+                    "ip",
+                    &["-6", "route", "add", &ip_net.to_string(), "dev", "wg_exit"],
```

Comment:
the verification might be better done at the level above this function where we can attribute the error to a given client. Also we should for sure print an error if we encounter this situation.

Right now if a client had a subnet error in the db they would just mysteriously not work. Printing an error here would help, but just say 'hey this subnet is broken'. But if we print an error in the user setup function then we'll be able to say what router with what subnet is having the issue.

Reply:
fixed